### PR TITLE
[MPP-1708] Popup Refactor - Masks Panel

### PR DESCRIPTION
**Pull request by maxxcrawford**
## Summary

Links:
- Jira Tickets:
  - https://mozilla-hub.atlassian.net/browse/MPP-1712
  - https://mozilla-hub.atlassian.net/browse/MPP-1713
  - https://mozilla-hub.atlassian.net/browse/MPP-1714
- Blocked by #461 
- L10N PR: https://github.com/mozilla-l10n/fx-private-relay-add-on-l10n/pull/44
- Epic PR #456 

Caveats/Not included: 
- Toggle/Expanded view of masks (to manage block level of masks) 
- Search bar/filter masks for premium masks
- Custom (subdomain'ed) masks generation for premium users

### Acceptance Criteria

[**MPP-1712**](https://mozilla-hub.atlassian.net/browse/MPP-1712)

- [x] If no aliases are created show placeholder screen: https://www.figma.com/file/NP5tLeIFJeHC5HvaqpLQdp?node-id=144:7504#184844686
- [x] For free users show alias count out of 5. 
- [x] Free and Premium users
- [x] Premium show all aliases that they have
- [x] Include loading animation 

[**MPP-1713**](https://mozilla-hub.atlassian.net/browse/MPP-1713)

- [x] Clicking copy button copies the alias to a user’s clipboard. 
- [x] A tag appears after a user clicks saying “Copied”

[**MPP-1714**](https://mozilla-hub.atlassian.net/browse/MPP-1714)

- [x] Random Mask creation 
- [x] Free and premium users
- [x] If free a user has less than 5 aliases, they should see a button for generate new alias which generates a new random alias
- [x] premium users should always be able to create a new alias 
- [x] Show new alias in the list of alias
- [x] show a tag that says “Mask created”
- [x] If a free user has 5 aliases then they should see an upsell button instead of the generate new alias button. 
- [x] If a free user has 5 aliases also show a red banner at the top of the add-on indicating they have reached their limit and should upgrade (see designs)

## Testing

TBD

## Screenshots

(Free Account) No masks created

image

Free account: 1 mask (with label set) 

image

Free account with maximum masks set (cannot generate more) 

image

Premium account (more than 5 more masks) 

image

### CODE REVIEW

File: src/css/popup-new.css [hunk 1]
```diff
@@ -34,6 +35,12 @@
   padding: var(--spacingMd);
 }
 
+/* Magic Number: Custom max-height for masks panel */
```
Comment (Vinnl) [inline src/css/popup-new.css:38]: This doesn't tell me anything about where the number came from :sweat_smile: wy not e.g. 400px?

File: src/css/popup-new.css [hunk 2]
```diff
@@ -42,6 +49,9 @@
   /* Bottom corners only */
   border-radius: 0 0 var(--borderRadiusSm) var(--borderRadiusSm);
   background-color: var(--colorWhite);
+  position: sticky;
+  top: 0;
+  z-index: 20;
```
Comment (Vinnl) [inline src/css/popup-new.css:54]: Ideally z-indexes are accompanied by a comment explaining what they're covering, so future contributors know whether they can be removed and/or what z-indexes to use for other elements.

**Review by Vinnl — APPROVED** [review]
We're still in the "just cursory high-level review of individual parts" phase, right?

File: src/css/popup-new.css (hunk 1 above)
Comment (maxxcrawford) [reply src/css/popup-new.css:38]: Heh. Revised comment to refactor to make it not an absolute number. It's going to break in future iterations. Leaving as a placeholder for now.

File: src/css/popup-new.css (hunk 2 above)
Comment (maxxcrawford) [reply src/css/popup-new.css]: Welp, good news. I went and tried it without a Z-index and it worked without anything. 💣 Revised in 4021b7f61fa14cabcfea18a7169a3c7ffcc1b4ab

File: src/css/popup-new.css (hunk 1 above)
Comment (maxxcrawford) [reply src/css/popup-new.css:39]: Revised in 4021b7f61fa14cabcfea18a7169a3c7ffcc1b4ab

**Pull request merged**